### ISSUE TITLE
ensure consistent description of pinned-domain-cert across doc & voucher doc

### ISSUE THREAD

**Issue by pritikin** [anima-wg/anima-bootstrap#63]

**Comment by pritikin** [comment]
The use of "domainID" in the log is an example of inconsistency here. Need to clarify that the domainID field is actually the hash of the public key etc of the pinned-domain-cert (whatever that is)

**Comment by pritikin** [comment]
First, a word on the voucher document itself. It specifies the pinned-domain-cert as: 
           This certificate is used by a pledge to trust a public key
           infrastructure, in order to verify a domain certificate
           supplied to the pledge separately by the bootstrapping
           protocol.  The domain certificate MUST have this certificate
           somewhere in its chain of certificates.  This certificate
           MAY be an end-entity certificate, including a self-signed
           entity."

The BRSKI document has plenty of room to mess this up so i'll go verify. Hold my beer...

**Comment by pritikin** [comment]
Ok. So there wasn't any normative problems but in verifying this I got confused (by section names) and ended up with a list of issues. They are detailed below (with their fixes).

RAW NOTES:
From voucher doc:

           This certificate is used by a pledge to trust a public key
           infrastructure, in order to verify a domain certificate
           supplied to the pledge separately by the bootstrapping
           protocol.  The domain certificate MUST have this certificate
           somewhere in its chain of certificates.  This certificate
           MAY be an end-entity certificate, including a self-signed
           entity.";


From BRSKI doc:

s2.1 behavior of a pledge

After step 4, the pledge has received and authenticated an explicit trust anchor (the pinned-domain-cert in the voucher response). A secure transport exists between pledge and registrar, and it may be used for things other than enrollment into a PKI.

FIXED: figure 2 includes a reference to appendixC that is out of date. This should be removed.
FIXED: the states are numbered in the list but not in figure2. Should use consistent labeling for clarity.
FIXED: 1st sentence of final paragraph is confusing and a repeat of the text on imprint above. Moved it to a more generic statement about the optional nature of EST integration. 

s2.5.5 PKI

The domain's registrar uses the "pinned-domain-cert" voucher field to distribute a trust anchor for authenticating itself to the pledge.

s2.5.1 lack of realtime clock

Once the voucher is accepted the validity period of the pinned-domain-cert in the voucher now serves as a valid time window. As explained in Section 5.4.4, the MASA has checked the registrar's certificate against real clocks , the endorsement of the MASA allows the pledge to treat the notBefore and notAfter dates as being constraints on any subsequent certificate validity periods that may need to be checked: for instance, validating peer certificates during ANIMA ACP setup.

s3.1 tree diagram

s3.3 yang module

      refine "voucher/pinned-domain-cert" {
        mandatory false;
      }

s5.4

FIXED: ending the section with a ":" isn't clear to the reader. 
SUGGESTION: make it a complete sentence as in: "The MASA performs the actions detailed in the following subsections before issuing a voucher".
FIXED: I consistently was getting lost on the subsections; there is nothing to distinguish
them. I updated the titles to all be in the form of something the MASA or pledge does.

s5.4.1

FIXED: clarify that this is something the MASA does when handling renew requests? (title change). 
FIXED: Use 'CMS' instead of "PKCS#7" (also made change in 5.2)

s5.4.2 voucher-request signature consistency

Performing this check provides value to domain PKI by assuring the domain administrator that the MASA service will only respect claims from authorized Registration Authorities of the domain.

FIXED: "value to the domain PKI" needs to the "the" and might be expanded.

"(The requirement for the registrar to include the Domain CA certificate in the signature structure was stated above.)" Where? Here: s5.4 "The registrar MUST sign the registrar voucher-request. The entire registrar certificate chain, up to and including the Domain CA, MUST be included in the PKCS#7 structure.". 
FIXED: now references s5.4 instead of just "above".

s5.4.1 renew for expired voucher

If the request is for a previous (expired) voucher using the same registrar (as determined by the registrar pinned-domain-cert) and the MASA has not been informed that the claim is invalid then the request for a renewed voucher SHOULD be automatically authorized.

s5.4.5 pledge proximity assertion

MAY verify that the registrar voucher-request includes the 'prior-signed-voucher' field populated with a signed pledge voucher-request that includes a 'proximity-registrar-cert' that is consistent with the certificate used to sign the registrar voucher-request.

QUESTION: there is a vague connection here between "consistent with" and the pinned cert but not sure if we need to change any language. 

s5.4.6 registrar pinning

The registrar's certificate chain is extracted from the signature method and the root certificate is used to populate the "pinned-domain-cert" of the voucher being issued. The domainID (e.g., hash of the root public key) is determined from the pinned-domain-cert and is used to update the audit log.
FIXED: Broke into two sentences and clarified the reference to the domain CA cert from s5.4

s5.5 voucher response

The 'pinned-domain-cert' element of the voucher contains the domain CA's public key. The pledge MUST use the 'pinned-domain-cert' trust anchor to immediately complete authentication of the provisional TLS connection.

The pledge MUST be prepared to parse and fail gracefully from a voucher response that does not contain a 'pinned-domain-cert' field. The pledge MUST be prepared to ignore additional fields that it does not recognize.

s5.5.1 completing authentication of provisional TLS connection

If a registrar's credentials cannot be verified using the pinned-domain-cert trust anchor from the voucher then the TLS connection is immediately discarded and the pledge abandons attempts to bootstrap with this discovered registrar.

The registrar MUST use a certificate that chains to the pinned-domain-cert as its TLS server certificate.

QUESTION: Do we need text here that points back to voucher doc text about "MAY be an end-entity certificate, including a self-signed entity"? 

The pinned-domain-cert is installed as an Explicit Trust Anchor for future operations. It can therefore can be used to authenticate any dynamically discovered EST server that contain the id-kp-cmcRA extended key usage extension as detailed in EST RFC7030 section 3.6.1; but to reduce system complexity the pledge SHOULD avoid additional discovery operations. Instead the pledge SHOULD communicate directly with the registrar as the EST server. The 'pinned-domain-cert' is not a complete distribution of the [RFC7030] section 4.1.3 CA Certificate Response, which is an additional justification for the recommendation to proceed with EST key management operations. Once a full CA Certificate Response is obtained it is more authoritative for the domain than the limited 'pinned-domain-cert' response.

5.8.1 EST Distribution of CA Certificates

This ensures that the pledge has the complete set of current CA certificates beyond the pinned-domain-cert (see Section 5.5.1 for a discussion of the limitations inherent in having a single certificate instead of a full CA Certificates response.) Although these limitations are acceptable during initial bootstrapping, they are not appropriate for ongoing PKIX end entity certificate validation.
